runhouse/resources/hardware/utils.py

```diff
@@ -571,4 +571,4 @@ def ssh_tunnel(
 
     # ssh_tunnel should certainlly be non-None at this point.
     cache_open_tunnel(address, ssh_port, ssh_tunnel)
-    return ssh_tunnel
+    return ssh_tunnel, local_port
```

Comment:
regarding the current testing errors, although you change it to `return ssh_tunnel, local_port` here, there are other break out points above where only `ssh_tunnel` is being returned, and it is likely running into one of those paths and unable to split up `ssh_tunnel` as a tuple